src/app/components/AssignmentForm.tsx

```diff
@@ -133,8 +132,8 @@ export default class AssignmentForm extends React.Component<AssignmentFormProps
                                         <ListGroupItem key={index}>
                                             <Button bsStyle="danger" onClick={() => fields.remove(index)} className="pull-right"><Glyphicon glyph="trash" /></Button><br />
                                             <Field name={`${recurrenceInfoFieldName}.days`} component={DaysOfWeekChecklist} label="Days" />
-                                            <Field name={`${recurrenceInfoFieldName}.startTime`} component={TimeField} label="Start Time" />
-                                            <Field name={`${recurrenceInfoFieldName}.endTime`} component={TimeField} label="End Time" />
+                                            <Field name={`${recurrenceInfoFieldName}.startTime`} component={(f)=><DateTimeField showDate={false} showTime closeOnSelect={false} {...f}/>} label="Start Time" />
```

Comment:
If you find that you're re-using this pattern a lot you could pull it out into an anonymous function type component i.e.:

```ts
const TimeField = (f:any)=><DateTimeField showDate={false} showTime closeOnSelect={false} {...f}/>
...
...
...
<Field component={TimeField} .../>
```
Just to avoid the repetition.  If this works and you liked the pattern that you had before.  You could still have a **TimeField** component.  But instead of the default export being a PureComponent, it would be this function.  You'd likely want to change the `(f:any)` to have the appropriate type info for the form field props.

The difference with this approach vs the previous.  Is that your consolidating all of the logic associated with onChange / values etc into one component.  Then you're consolidating the logic / display appearance of a "TimeField" / "DateField" into a component that is "composed" from the same component (just with different properties set)